### PR TITLE
Support moving pieces

### PR DESCRIPTION
**Pull request by dennisgit42**
My first attempt, having a lot of problems getting pieces/show.html.erb to recognize the piece and square the user selected. Any feedback is welcome.

### CODE REVIEW

**Comment by dennisgit42** [comment]
I am having a lot of problem figuring out how to use the view pieces/show.html.erb to pass the coords of the destination square to the pieces#update action so that clicking a link in the destination square will update the location of the piece.

**Comment by dennisgit42** [comment]
Think I figured out the destination coords problem, now just re-writing most of games#show and pieces#show to include the necessary links.

**Comment by dennisgit42** [comment]
So all tests now passing, just need Mike to sign off and I can merge into master

File: app/models/game.rb [hunk 1]
```diff
@@ -41,4 +41,9 @@ def populate_board!
     Piece.create(game_id: id, is_white: true, type: KING, x_position: 5, y_position: 8)
     Piece.create(game_id: id, is_white: true, type: QUEEN, x_position: 4, y_position: 8)
   end
+
+  def render_piece(x, y)
+    piece = Piece.find_by(game_id: id, x_position: x, y_position: y)
```
Comment (mscoutermarsh) [inline app/models/game.rb]: This could be written as

`pieces.find_by(x_position: x, y_position: y)`

**Pull request merged**